resolver-dns/src/main/java/io/netty/resolver/dns/AuthoritativeDnsServerCache.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2018 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.resolver.dns;
+
+import io.netty.channel.EventLoop;
+import io.netty.util.internal.UnstableApi;
+
+import java.net.InetSocketAddress;
+import java.util.List;
+
+/**
+ * Cache which stores the nameservers that should be used to resolve a specific hostname.
+ */
+@UnstableApi
+public interface AuthoritativeDnsServerCache {
+
+    /**
+     * Returns the cached nameservers that should be used to resolve the given hostname. The returned unmodifiable
+     * {@link List} may contain unresolved {@link InetSocketAddress}es that will be resolved when needed while
+     * resolving other domain names.
+     *
+     * @param hostname the hostname
```

Comment:
This should indicate if it returns `null` or an empty list if the host isn't found.

Reply:
sounds good.

Reply:
actually let me just change it to always return an empty List. This makes it also easier to handle.

Reply:
done